app/templates/layouts/_base.html

```diff
@@ -133,4 +133,26 @@
       document.write("<script async nonce='{{ csp_nonce() }}' src='http://HOST:5075/browser-sync/browser-sync-client.js?v=2.24.4'><\/script>".replace("HOST", location.hostname));
     //]]></script>
     {% endif %}
+
+    <script nonce="{{ csp_nonce() }}" >
+        (function() {
+          var printed = 0;
+          // FF and IE
+          var onbeforePrinting = function() {
+            if (printed == 0) {
+                window.dataLayer.push({ 'event': 'onBeforePrint', 'currentURL': location.pathname });
+                printed = 1;
+            }
+          };
+          // matchMedia Support
+          if (window.matchMedia) {
+              var mediaQueryList = window.matchMedia('print');
+              mediaQueryList.addListener(function(mql) {
+                  if (mql.matches)
+                      onbeforePrinting();
+              });
+          }
+          window.onbeforeprint = onbeforePrinting;
+        }());
```

Comment:
This should be moved into a file inside the app assets, so `app/assets/js/app/modules/`. Otherwise, this would not get transpiled by Babel. Once done you would also be able to use ES6 syntax like `let` instead of `var` which Babel will transpile when running `yarn compile`.

Reply:
Now added to JS module